### PR TITLE
feat: MVP include un-persisted results from the ingester in query results

### PR DESCRIPTION
**Pull request by alamb**
~Draft / WIP / YOLO PR: once I get it kind of working I will split off particular parts into their own PRs and list below.~

# Rationale
The first "could minimally claim to work" PR for https://github.com/influxdata/influxdb_iox/issues/3996

See design doc: https://docs.google.com/document/d/1LJZyw-Ik3_EMpXwEFHu3a9c2RJIR3uxpnDmEfL56Sl4/edit#heading=h.fe76eikpw09c

This PR allows the querier to include un-persisted results from the ingester to drastically reduce "time to be readable" 

# Changes
1. Make a flight call from querier to ingester
2. Implement query interface for`IngesterPartition` for data returned from the querier 
3. (Very) basic end to end test coverage

# PR sequence leading up to this one
- [x] Add `Error` back to `QueryDatabase::chunks` trait (so ingester_connection can return error): https://github.com/influxdata/influxdb_iox/pull/4260
- [x]  Wire in basic "specify ingester address" configuration: https://github.com/influxdata/influxdb_iox/pull/4259
- [x] Support multiple ingester addresses https://github.com/influxdata/influxdb_iox/pull/4271
- [x] Clean up flight APIs https://github.com/influxdata/influxdb_iox/pull/4273

# Follow on tasks / PRs:
- [ ] Update ingester request to querier to ask about all ingester ids (rather than just one)
- [ ] Update ingester response to querier to include data per partition rather than data combined for all partitions
- [ ] Implement  logic that detects when the max persisted data is inconsistent with known parquet
- [ ] Do something useful (like retry) when inconsistency is hit
- [ ] File ticket for why the `RecordBatches` batches sent from the ingester are not Dictionary encoded: https://github.com/influxdata/influxdb_iox/issues/4275

### CODE REVIEW

File: querier/src/ingester/mod.rs [hunk 1]
```diff
@@ -106,10 +139,45 @@ impl IngesterConnection for IngesterConnectionImpl {
                 .await
                 .context(HandshakeSnafu { ingester_address })?;
 
-            // TODO Coming Soon: create the actual IngesterPartition and return them here.
-            partitions.extend(vec![]);
+            let ingester_query_request = IngesterQueryRequest {
```
Comment (alamb) [inline querier/src/ingester/mod.rs:132]: here is more plumbing to make the actual call / response -- clearly there is a lot more to do

File: querier/src/ingester/mod.rs [hunk 2]
```diff
@@ -128,82 +196,266 @@ impl IngesterConnection for IngesterConnectionImpl {
...
+        table_name: Arc<str>,
+        expected_schema: Arc<Schema>,
+        parquet_max_sequence_number: Option<SequenceNumber>,
+        batches: Vec<RecordBatch>,
+    ) -> Result<Self> {
+        // ensure that the schema of the batches matches the required output schema by CAST'ing type
+        // TODO needed because flight client doesn't seem to send dictionaries??
+        //
+        // Also todo: maybe make this part of IngesterPartition (so it can be tested)
+        let batches = batches
```
Comment (alamb) [inline querier/src/ingester/mod.rs:219]: This conversion / test is the result of hard earned knowledge debugging last week

File: querier/src/ingester/mod.rs [hunk 3]
```diff
@@ -128,82 +186,253 @@ impl IngesterConnection for IngesterConnectionImpl {
...
 
     fn table_name(&self) -> &str {
-        todo!()
+        self.table_name.as_ref()
     }
 
     fn may_contain_pk_duplicates(&self) -> bool {
-        todo!()
+        // who knows what is in here!
+        true
```
Comment (crepererum) [inline querier/src/ingester/mod.rs:276]: I thought the ingester runs dedup before creating the record batches?

File: querier/src/ingester/mod.rs [hunk 4]
```diff
@@ -128,82 +186,253 @@ impl IngesterConnection for IngesterConnectionImpl {
...
     }
 
     fn chunk_type(&self) -> &str {
-        todo!()
+        "IngesterPartition"
+    }
+
+    fn order(&self) -> ChunkOrder {
+        // TODO a real order
+        ChunkOrder::new(1).unwrap()
```
Comment (crepererum) [inline querier/src/ingester/mod.rs:343]: Since this is always the last chunk, I think you can safely use `ChunkOrder::new(u32::MAX).unwrap()` here.

**Review by crepererum — APPROVED** [review]

File: querier/src/ingester/mod.rs (hunk 3 above)
Comment (alamb) [reply querier/src/ingester/mod.rs:276]: 👍

File: querier/src/ingester/mod.rs (hunk 4 above)
Comment (alamb) [reply querier/src/ingester/mod.rs:343]: will change

**Comment by alamb** [comment]
Addressed comments and rebased to fix conflicts

**Pull request merged**